bin/lib/lock_webtask.js

```diff
@@ -0,0 +1,155 @@
+        </script>
+      </body>
+    </html>
+`);
+
+const FUNC_TO_RUN = function() {
+    var lock = new Auth0Lock(CLIENT_ID, AUTH0_DOMAIN);
+    var lock_opts = {
+        authParams: {
+            scope: 'openid email'
+        }
+    };
+
+    lock.show(lock_opts, function (err, profile, id_token) {
+        if(err) return console.log(err);
+
+        var opts = {
+            headers: {
+                'Authorization': 'Bearer ' + id_token
+            }
+        };
+
+        request(REQ_METHOD, TASK_URL, opts)
+            .then(function (res) {
+                switch(res.headers['content-type']) {
+                    case 'application/json':
+                        var parsed = JSON.parse(res.body);
+
+                        document.body.innerHTML = '<pre>' + JSON.stringify(parsed, null, 1) + '</pre>';
+                        break;
+                    default:
+                        document.body.innerHTML = res.body;
+                }
+            })
+            .catch(function (err) {
+                console.log(err);
+            });
+    });
+}.toString();
+
+function getAuthToken (req) {
+    if(req.query.key)
+        return req.query.key;
+
+    if(req.headers.authorization)
+        return req.headers.authorization
+            .match(/^bearer\s+(.+)$/i)[1];
+
+    return false;
+}
+
+const query_delete = [
+    'baseUrl',
+    'container',
+    'taskname',
+    'domain',
+    'clientId',
+];
+
+app.get('*', (req, res) => {
+    const ctx = req.webtaskContext;
+
+    if(!ctx.data.container || !ctx.data.taskname) {
+       res.statusCode = 400;
+       res.end('Must provide container & taskname params');
+    }
+
+    if(!ctx.data.clientId && !ctx.data.domain) {
+       res.statusCode = 400;
+       res.end('Must provide clientId & domain params');
+    }
+
+    let query = qs.parse(
+        url.parse(req.url).query
+    );
+
+    for(let i in query)
+        if(i.match('webtask_') && i !== 'webtask_no_cache')
+            delete query[i];
+
+    query_delete
+        .forEach(function (key) {
+            delete query[key];
+        });
+
+    const TASK_URL = (ctx.data.baseUrl   || 'https://webtask.it.auth0.com') +
+                   '/api/run/'                 +
+                   ctx.data.container          +
+                   '/'                         +
+                   ctx.data.taskname           +
+                   url.parse(req.url).pathname +
+                   '?' + qs.stringify(query);
+
+    const AUTH_TOKEN = getAuthToken(req);
+
+    if(AUTH_TOKEN)
+        request({
+            method: req.method,
+            url: TASK_URL,
+            headers: {
+                Authorization: 'Bearer ' + AUTH_TOKEN
+            }
+        }).pipe(res);
+    else
+        res.end(
+            VIEW({
+                REQ_METHOD: req.method,
+                CLIENT_ID: ctx.data.clientId,
+                AUTH0_DOMAIN: ctx.data.domain,
+                FUNC_TO_RUN,
```

Comment:
I think I prefer it outside of the HTML because it makes editing/maintaining easier. I gave it a better name, but if you feel strongly there's no reason it can't be embedded in the HTML string.